lua/core/mappings.lua

```diff
@@ -300,18 +338,19 @@ end
 if is_available "nvim-colorizer.lua" then
   maps.n["<leader>uC"] = { "<cmd>ColorizerToggle<cr>", desc = "Toggle color highlight" }
 end
-maps.n["<leader>uS"] = { function() astronvim.ui.toggle_conceal() end, desc = "Toggle conceal" }
 maps.n["<leader>ud"] = { function() astronvim.ui.toggle_diagnostics() end, desc = "Toggle diagnostics" }
 maps.n["<leader>ug"] = { function() astronvim.ui.toggle_signcolumn() end, desc = "Toggle signcolumn" }
 maps.n["<leader>ui"] = { function() astronvim.ui.set_indent() end, desc = "Change indent setting" }
 maps.n["<leader>ul"] = { function() astronvim.ui.toggle_statusline() end, desc = "Toggle statusline" }
+maps.n["<leader>uL"] = { function() astronvim.ui.toggle_codelens() end, desc = "Toggle CodeLens refresh" }
 maps.n["<leader>un"] = { function() astronvim.ui.change_number() end, desc = "Change line numbering" }
-maps.n["<leader>us"] = { function() astronvim.ui.toggle_spell() end, desc = "Toggle spellcheck" }
+maps.n["<leader>uN"] = { function() astronvim.ui.toggle_ui_notifications() end, desc = "Toggle UI notifications" }
 maps.n["<leader>up"] = { function() astronvim.ui.toggle_paste() end, desc = "Toggle paste mode" }
+maps.n["<leader>us"] = { function() astronvim.ui.toggle_spell() end, desc = "Toggle spellcheck" }
+maps.n["<leader>uS"] = { function() astronvim.ui.toggle_conceal() end, desc = "Toggle conceal" }
 maps.n["<leader>ut"] = { function() astronvim.ui.toggle_tabline() end, desc = "Toggle tabline" }
 maps.n["<leader>uu"] = { function() astronvim.ui.toggle_url_match() end, desc = "Toggle URL highlight" }
 maps.n["<leader>uw"] = { function() astronvim.ui.toggle_wrap() end, desc = "Toggle wrap" }
 maps.n["<leader>uy"] = { function() astronvim.ui.toggle_syntax() end, desc = "Toggle syntax highlight" }
-maps.n["<leader>uN"] = { function() astronvim.ui.toggle_ui_notifications() end, desc = "Toggle UI notifications" }
 
-astronvim.set_mappings(astronvim.user_plugin_opts("mappings", maps))
+astronvim.set_mappings(astronvim.user_opts("mappings", maps))
```

Comment:
I am not for/against for `mappings.lua`, is there a reason to not use `lazy.nvim`, https://github.com/folke/lazy.nvim#:~:text=load%20on%20filetype-,keys,-string%3F%20or?

Reply:
There are a lot of reasons here mainly regarding ease of control to the user. They don't need to spend time searching through source code to figure out which plugin binds a mapping, if they don't want one they can set it to false in one table. Also it keeps the user from safely overriding mappings in a methodological way. If the user defines a mapping in the mappings table you know it will only be set once and will be safe. With lazy it will be down to what plugin is loaded first. There are a lot more reasons for this as well, but those are a few.

TL;DR this increases usability, ease of configuration to the user, stability, and safety by an extremely large margin